test/functional/lua/ui_spec.lua

```diff
@@ -130,4 +130,44 @@ describe('vim.ui', function()
     end)
 
   end)
+  describe('confirm', function ()
+    it('can choose item', function ()
+      local result = exec_lua[[
+        local opts = {
+            choices = {"Foo", "Bar"},
+	    default = 1,
+        }
+	local ch = nil
+	local on_confirm = function(idx, choice)
+	  ch = choice
+	end
+
+	vim.ui.confirm("Choice?", opts, on_confirm)
+        vim.wait(100, function() return ch ~= nil end)
```

Comment:
Try using `vim.api.nvim_feedkeys()` here to feed in a choice rather than `vim.wait()`.